### PR TITLE
Signals Episode VI: Return of the XenoMacros

### PR DESCRIPTION
**Pull request by spookydonut**
alt title: Refactors abilities to work with keybinds

## About The Pull Request
keybinds are great, good work PsyKzz, this lets players bind literally every single xeno ability to a key.

## Why It's Good For The Game
lets you keybind abilities again.

## Changelog
:cl:
add: All xeno abilities can be used with keybinds now, although for targeted abilities the keybind will only select it.
/:cl:

### CODE REVIEW

**Comment by spookydonut** [comment]
basically done apart from like filling in the rest of the descriptions and names

**Comment by spookydonut** [comment]
Theres issues with this where multiple keybinds using the same key does weird things

File: code/datums/keybinding/xeno.dm [hunk 1]
```diff
@@ -1,26 +1,407 @@
...
+	var/ability_signal
 
+/datum/keybinding/xeno/down(client/user)
+	if(SEND_SIGNAL(user.mob, ability_signal) & COMSIG_XENOABILITY_HAS_ABILITY)
+		return TRUE
+
+	if(!isxeno(user.mob))
+		return
+
+	to_chat(user, "<span class='notice'>You don't have this ability.</span>") // TODO Is this spammy?
```
Comment (ninjanomnom) [inline code/datums/keybinding/xeno.dm]: yes

Comment (spookydonut) [reply code/datums/keybinding/xeno.dm]: it also made sharing keys for different abilities not work

**Pull request merged**